### PR TITLE
[Keyboard] New Keyboard - BB Steno

### PR DESCRIPTION
**Pull request by kylemccreery**
## Description

Initial add of MechWild's BB Steno. A simple steno keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mechwild/bbs/readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+![BBS](https://i.imgur.com/XIjlzKOh.png)
+
+A bare-bones stenography keyboard. No bells or whistles. Simple, cheap, effective, steno.
+
+* Keyboard Maintainer: [Kyle McCreery](https://github.com/kylemccreery)
+* Hardware Supported: BBS v0.1
+* Hardware Availability: [BBS on MechWild](https://mechwild.com/product/bb-steno/)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mechwild/bbs/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mechwild/bbs/bbs.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            if(active) { tap_code(KC_CLCK); }
+            break;
+        break;
+    }
+    return true;
+}
+#endif
+
+void eeconfig_init_kb() {
+    steno_set_mode(STENO_MODE_GEMINI); // or STENO_MODE_BOLT
```
Comment (drashna) [inline keyboards/mechwild/bbs/bbs.c:20]:
```suggestion
    steno_set_mode(STENO_MODE_GEMINI); // or STENO_MODE_BOLT
    eeconfig_init_user();
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/mechwild/bbs/info.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (fauxpark) [inline keyboards/mechwild/bbs/info.json:1]: image

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mechwild/bbs/info.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    "layouts": {
+        "LAYOUT": {
+            "layout": [{"label":"k03", "x":3, "y":0.13}, {"label":"k08", "x":8.75, "y":0.13}, {"label":"k02", "x":2, "y":0.25}, {"label":"k04", "x":4, "y":0.25}, {"label":"k05", "x":5, "y":0.25}, {"label":"k06", "x":6.75, "y":0.25}, {"label":"k07", "x":7.75, "y":0.25}, {"label":"k09", "x":9.75, "y":0.25}, {"label":"k00", "x":0, "y":0.5}, {"label":"k01", "x":1, "y":0.5}, {"label":"k0A", "x":10.75, "y":0.5}, {"label":"k0B", "x":11.75, "y":0.5}, {"label":"k13", "x":3, "y":1.1300000000000001}, {"label":"k18", "x":8.75, "y":1.1300000000000001}, {"label":"k12", "x":2, "y":1.25}, {"label":"k14", "x":4, "y":1.25}, {"label":"k15", "x":5, "y":1.25}, {"label":"k16", "x":6.75, "y":1.25}, {"label":"k17", "x":7.75, "y":1.25}, {"label":"k19", "x":9.75, "y":1.25}, {"label":"k10", "x":0, "y":1.5}, {"label":"k11", "x":1, "y":1.5}, {"label":"k1A", "x":10.75, "y":1.5}, {"label":"k1B", "x":11.75, "y":1.5}, {"label":"k25", "x":4.75, "y":3, "h":1.5}, {"label":"k26", "x":7, "y":3, "h":1.5}, {"label":"k23", "x":2.75, "y":3.25, "h":1.5}, {"label":"k24", "x":3.75, "y":3.25, "h":1.5}, {"label":"k27", "x":8, "y":3.25, "h":1.5}, {"label":"k28", "x":9, "y":3.25, "h":1.5}]
```
Comment (noroadsleft) [inline keyboards/mechwild/bbs/info.json:7]: Corrects the key sequence.

```suggestion
            "layout": [
                {"label":"k00", "x":0, "y":0.5},
                {"label":"k01", "x":1, "y":0.5},
                {"label":"k02", "x":2, "y":0.25},
                {"label":"k03", "x":3, "y":0.125},
                {"label":"k04", "x":4, "y":0.25},
                {"label":"k05", "x":5, "y":0.25},
                {"label":"k06", "x":6.75, "y":0.25},
                {"label":"k07", "x":7.75, "y":0.25},
                {"label":"k08", "x":8.75, "y":0.125},
                {"label":"k09", "x":9.75, "y":0.25},
                {"label":"k0A", "x":10.75, "y":0.5},
                {"label":"k0B", "x":11.75, "y":0.5},

                {"label":"k10", "x":0, "y":1.5},
                {"label":"k11", "x":1, "y":1.5},
                {"label":"k12", "x":2, "y":1.25},
                {"label":"k13", "x":3, "y":1.125},
                {"label":"k14", "x":4, "y":1.25},
                {"label":"k15", "x":5, "y":1.25},
                {"label":"k16", "x":6.75, "y":1.25},
                {"label":"k17", "x":7.75, "y":1.25},
                {"label":"k18", "x":8.75, "y":1.125},
                {"label":"k19", "x":9.75, "y":1.25},
                {"label":"k1A", "x":10.75, "y":1.5},
                {"label":"k1B", "x":11.75, "y":1.5},

                {"label":"k23", "x":2.75, "y":3.25, "h":1.5},
                {"label":"k24", "x":3.75, "y":3.25, "h":1.5},
                {"label":"k25", "x":4.75, "y":3, "h":1.5},
                {"label":"k26", "x":7, "y":3, "h":1.5},
                {"label":"k27", "x":8, "y":3.25, "h":1.5},
                {"label":"k28", "x":9, "y":3.25, "h":1.5}
            ]
```

File: keyboards/mechwild/bbs/rules.mk [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = no        # Mouse keys
+EXTRAKEY_ENABLE = no        # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = no         # Encoder Enabled
+OLED_ENABLE = no            # OLED Enabled
```
Comment (noroadsleft) [inline keyboards/mechwild/bbs/rules.mk:19]:
```suggestion
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**